### PR TITLE
Enable parallel testing

### PR DESCRIPTION
**Pull request by wcgcyx**

### CODE REVIEW

File: messaging/txroot-transfer/src/test/java/net/consensys/gpact/txroot/TxRootAddTest.java [hunk 1]
```diff
@@ -215,41 +217,45 @@ private void proveTxReceipt(
...
-      // Show the receipts root that has been included in the block.
-      assert (receipts[i] != null && receipts[i].getBlockHash() != null);
-      // TODO check that all transactions ended up in the same block, and hence have the same block
-      // hash
+    while (true) {
+      for (int i = 0; i < numTransactionsPerBlock; i++) {
+        String privateKey0 = new KeyPairGen().generateKeyPairGetPrivateKey();
+        creds[i] = Credentials.create(privateKey0);
+  
+        TransactionReceiptProcessor txrProcessor =
```
Comment (ermyas) [inline messaging/txroot-transfer/src/test/java/net/consensys/gpact/txroot/TxRootAddTest.java:226]: nit: does this `txProcessor` need to be created for each transaction manager instance? It looks like a single instance can be reused?

File: messaging/txroot-transfer/src/test/java/net/consensys/gpact/txroot/TxRootAddTest.java [hunk 2]
```diff
@@ -215,41 +217,45 @@ private void proveTxReceipt(
...
+            TestEvents.load(testContractAddress, this.web3j, tms[i], this.freeGasProvider);
+      }
+  
+      BigInteger id = BigInteger.TWO;
+  
+      //    CompletableFuture<TransactionReceipt>[] transactionReceiptCompletableFutures =
+      //        (CompletableFuture<TransactionReceipt>[]) new Object[numTransactionsPerBlock];
+      CompletableFuture<?>[] transactionReceiptCompletableFutures =
+          new CompletableFuture<?>[numTransactionsPerBlock];
+      for (int i = 0; i < numTransactionsPerBlock; i++) {
```
Comment (ermyas) [inline messaging/txroot-transfer/src/test/java/net/consensys/gpact/txroot/TxRootAddTest.java:240]: nit: is there a reason why this second for-loop is separate from the first?

**Review by ermyas — APPROVED** [review]
LGTM. Minor comments added.

File: .circleci/config.yml [hunk 3]
```diff
@@ -205,8 +205,7 @@ jobs:
       - run: sleep 30
       - run:
           name: Build and test
-          command: ./gradlew test
-          # command: ./gradlew --parallel test
+          command: ./gradlew --parallel test
```
[the remark this answers is not in the document]
Comment (wcgcyx) [reply .circleci/config.yml:208]: sounds like a good idea, I'll update the PR

File: messaging/txroot-transfer/src/test/java/net/consensys/gpact/txroot/TxRootAddTest.java (hunk 1 above)
Comment (wcgcyx) [reply messaging/txroot-transfer/src/test/java/net/consensys/gpact/txroot/TxRootAddTest.java:226]: That's true. Thank you for pointing it out. I'll update the PR.

**Pull request merged**